server/interfaces/interfaces.go

```diff
@@ -956,6 +956,21 @@ type RunnerPool interface {
 	GetBuildRoot() string
 }
 
+// CommandRunner is an interface for running command-line commands.
+type CommandRunner interface {
+	Run(ctx context.Context, command *repb.Command, workDir string, statsListener func(*repb.UsageStats), stdio *Stdio) *CommandResult
+}
```

Comment:
This naming is a bit unfortunate since we already have something called `commandRunner` in `runner.go`. We could either rename the existing `runner` to something better or call this something like `ProcessRunner` (since ultimately `Run()` here is doing low-level syscalls to create a unix/windows process and get its result, whereas the existing runner is doing higher-level stuff like managing workspaces and containers for a command).

Reply:
Yeah, I saw that, I think the most correct thing to do would be to rename the `runner.go` version to `taskRunner`, as that's more consistent with the language in there and in the remote execution API. WDYT?

Reply:
`taskRunner` SGTM